app/assets/javascripts/es6.js

```diff
@@ -0,0 +1,23 @@
+class Animal {
+  constructor(name) {
+    this.name = name;
+  }
+
+  speak() {
+    console.log(this.name + ' makes a noise.');
+  }
+}
+
+class Dog extends Animal {
+  speak() {
+    console.log(this.name + ' barks.');
+  }
+}
+
+class Cat extends Animal {
+  speak() {
+    console.log(this.name + " loves to chase cards and bite their tires like a dog, but isn't one.");
+  }
+}
+
+a(foo => { if (true) {}; });
```

Comment:
"a" is not defined.<br>Expected parentheses around arrow function argument.<br>"foo" is defined but never used<br>Unexpected constant condition.<br>Empty block statement.<br>Unnecessary semicolon.